ft_itoa.c

```diff
@@ -0,0 +1,48 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_itoa.c                                          :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: bchristo <bchristo@student.42wolfsburg.    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2021/05/27 03:00:41 by bchristo          #+#    #+#             */
+/*   Updated: 2021/05/27 15:31:27 by bchristo         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "libft.h"
+
+static int	*helperPlus(char *p, int *n)
+{
+	*--p = '0' + (*n % 10);
+	*n /= 10;
+	return (n);
+}
+
+static int	*helperMinus(char *p, int *n)
+{
+	*--p = '0' - (*n % 10);
+	*n /= 10;
+	return (n);
+}
+
+char	*ft_itoa(int n)
+{
+	static char	buf[19 + 2];
+	char		*p;
+
+	p = buf + 19 + 1;
+	if (n >= 0)
+	{
+		while (*helperPlus(p--, &n))
+			continue ;
+		return (p);
```

Comment:
Whit this return the returning pointer is not "allocated with malloc". I don't know if it will cause a problem. But it's strange that in plus case this static buffer is returned and in minus case a dynamic allocated memory is returned.